geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/ZAddParameterRequirements.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.ParameterRequirements;
+
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_INVALID_ZADD_OPTION_NX_XX;
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_NOT_A_VALID_FLOAT;
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_SYNTAX;
+
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.geode.redis.internal.data.ByteArrayWrapper;
+import org.apache.geode.redis.internal.netty.Command;
+import org.apache.geode.redis.internal.netty.ExecutionHandlerContext;
+
+public class ZAddParameterRequirements implements ParameterRequirements {
+  @Override
+  public void checkParameters(Command command, ExecutionHandlerContext context) {
+    int numberOfArguments = command.getProcessedCommand().size();
+
+    if (numberOfArguments < 4) {
+      throw new RedisParametersMismatchException(command.wrongNumberOfArgumentsErrorMessage());
+    }
+
+    int optionsFoundCount = confirmKnownSubcommands(command);
+
+    if ((numberOfArguments - optionsFoundCount - 2) % 2 != 0) {
+      throw new RedisParametersMismatchException(ERROR_SYNTAX);
+    }
+  }
+
+  private int confirmKnownSubcommands(Command command) {
+    int optionsFoundCount = 0;
+    boolean nxFound = false, xxFound = false, gtFound = false, ltFound = false;
+
+    List<ByteArrayWrapper> commandElements = command.getProcessedCommandWrappers();
```

Comment:
Can we implement this without using `ByteArrayhWrapper`, since we want to get rid of it?

Reply:
Yup, there's a new getProcessedCommand() call I can use. Just pushed up an update. Good catch!